src/IdealizedSimulations/neverworld_simulation.jl

```diff
@@ -0,0 +1,298 @@
+    x₂, y₂ = line.p₂.x, line.p₂.y
+
+    # Line segment vector components
+    ℓˣ = x₂ - x₁
+    ℓʸ = y₂ - y₁
+
+    # Fractional increment to closest segment point
+    ϵ = ((x - x₁) * ℓˣ + (y - y₁) * ℓʸ) / (ℓˣ^2 + ℓʸ^2)
+    ϵ = clamp(ϵ, 0, 1)
+
+    # Closest segment point
+    x′ = x₁ + ϵ * Δx
+    y′ = y₁ + ϵ * Δy
+    
+    return sqrt((x′ - x)^2 + (y′ - y)^2)
+end
+
+function distance(point::Point, line::Line)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+    num = abs((x₂ - x₁) * (y₁ - y) - (y₂ - y₁) * (x₁ - x))
+    den = sqrt((x₂ - x₁)^2 + (y₂ - y₁)^2)
+    return num / den
+end
+
+#####
+##### Bathymetry
+#####
+
+struct NeverworldBathymetry{G, B} <: Function
+    grid :: G
+    coastline_spline :: B
+    rim_width :: Float64
+    slope_width :: Float64
+    shelf_width :: Float64
+    shelf_depth :: Float64
+    abyssal_depth :: Float64
+    southern_channel_boundary :: Float64
+    northern_channel_boundary :: Float64
+end
+
+function NeverworldBathymetry(grid;
+                              abyssal_depth = 4000,
+                              shelf_depth = 200,
+                              shelf_width = 2.5,
+                              rim_width = shelf_width / 8,
+                              slope_width = shelf_width,
+                              southern_channel_boundary = -60,
+                              northern_channel_boundary = -40)
+
+    # Use grid spacing for "beach width"
+    Δ = max(grid.Δλᶠᵃᵃ, grid.Δφᵃᶠᵃ)
+
+    # Construct a cubic spline of the form `basin_depth(r)`, representing
+    # the "basin component" of the Neverworld bathymetry where `r` is the distance
+    # to the edge of the Neverworld (with units of degrees).
+    r_coast = Δ
+    r_beach = Δ + rim_width
+    r_mid_shelf = Δ + rim_width + shelf_width / 2
+    r_shelf = Δ + rim_width + shelf_width
+    r_abyss = Δ + rim_width + shelf_width + slope_width
+
+    Nx, Ny, Nz = size(grid)
+    x_max = xnode(Nx+1, 1, 1, grid, f, c, c) - xnode(1, 1, 1, grid, f, c, c)
+    y_max = ynode(1, Ny+1, 1, grid, c, f, c) - ynode(1, 1, 1, grid, c, f, c)
+    r_max = max(x_max, y_max)
+
+    basin_rim_distances = [0, r_coast,     r_beach,  r_mid_shelf,     r_shelf,       r_abyss,         r_max]
+    basin_depths        = [0, 0,       shelf_depth,  shelf_depth, shelf_depth, abyssal_depth, abyssal_depth]
+    coastline_spline = CubicSplineFunction{:x}(basin_rim_distances, basin_depths)
+
+    return NeverworldBathymetry(grid,
+                                coastline_spline,
+                                Float64(rim_width),
+                                Float64(slope_width),
+                                Float64(shelf_width),
+                                Float64(shelf_depth),
+                                Float64(abyssal_depth),
+                                Float64(southern_channel_boundary),
+                                Float64(northern_channel_boundary))
```

Comment:
Should these be same type as `eltype(grid)`? Perhaps if we have:

```julia
struct NeverworldBathymetry{FT, G, B} <: Function
    grid :: G
    coastline_spline :: B
    rim_width :: FT
    slope_width :: FT
    shelf_width :: FT
    shelf_depth :: FT
    abyssal_depth :: FT
    southern_channel_boundary :: FT
    northern_channel_boundary :: FT

    function NeverworldBathymetry(grid, coastline_spline, rim_width,
                                  slope_width, shelf_width, shelf_depth, abyssal_depth,
                                  southern_channel_boundary, northern_channel_boundary)

        FT = eltype(grid)
        B = typeof(coastline_spline)
        G = typeof(grid)

        return new{FT, G, B}(grid, coastline_spline, rim_width,
                            slope_width, shelf_width, shelf_depth, abyssal_depth,
                            southern_channel_boundary, northern_channel_boundary)
    end
end
```

Reply:
The bathymetry is recomputed to be `eltype` grid so it doesn't matter

Reply:
unless i suppose we do higher precision than `Float64`

Reply:
Ok!